src/shared/services/json-store.service.spec.ts

```diff
@@ -86,4 +86,79 @@ describe('JsonStoreService', () => {
     service.setIn(path, value);
   });
 
+  it('should shift the first element if the new pushed element exceeds the maximum history length', () => {
+    let json = fromJS({
+      aMap: {
+        aList: [
+          {
+            innerList: [
+              {innerProperty: '0'}
+            ]
+          }
+        ]
+      }
+    });
+    service.setJson(json);
+    let path = ['aMap', 'aList', 0, 'innerList', 0, 'innerProperty'];
+    for (let i = 1; i <= service.maxStateRestoreCounter; i++) {
+      service.addJsonToHistory();
+      service.setIn(path, `${i}`);
```

Comment:
using `setJson` would be better not to depend on `setIn` in this test because it has its own tests.